File: 0881-boats-to-save-people/0881-boats-to-save-people.cpp
class Solution {
public:
    /*
    Algorithm
    1)declare 2 ptrs one starting from 0 & the other from n-1
    2) sort
    3) while l<=r
    4) add elements on the l & r indexes if <= limit move both ptrs
    5) if not move only r ptr
    4) and increament count each time
    
    
    */
   int numRescueBoats(vector<int>& people, int limit) {
        int  l=0,counts=0;
        int n = people.size(), r= n-1;
        sort(people.begin(), people.end());
        while(l<=r){
           if(people[l] + people[r]<= limit){//if the sum <= limit we move both pointers
               l++;
               r--;
           }
            else{//if > move only r pointer
                r--;
            }
            counts++;
        }
        return counts;
    }
};

File: 0881-boats-to-save-people/0881-boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        end = len(people) - 1
        start = 0
        count = 0
        while start <= end:
            if people[start] + people[end] <= limit:
                count += 1
                start += 1
                end -= 1
            elif people[end] <= limit:
                count += 1
                end -= 1
                
        return count